proposals/0310-PerformInteractionMultipick.md

```diff
@@ -1,251 +1,232 @@
-
-          Only is valid if general result is "success:true".</description>
-    </param>
-  </function>
-```
+# Perform Interaction Multipick
+
+* Proposal: [SDL-0310](0310-PerformInteractionMultipick.md)
+* Author: [Michael Crimando](https://github.com/MichaelCrimando)
+* Status: **Returned for Revisions**
+* Impacted Platforms: [HMI / Core / iOS / Java Suite / JavaScript / RPC ]
+
+## Introduction
+
+For when a user wants to order something from an app, the user may want to pick 1 or more of their favorite items from a list. This would allow that. Currently, perform interaction only accepts 1 input and clears.
+
+## Motivation
+
+As a user, I want to be able to pick multiple things from a list of my favorite orders so that I can order everything I want through the head unit.
+
+Here's an example from an overall plan
+
+<img src="../assets/proposals/0310-PerformInteraction-Multipick/PerformInteraction Multipick overview.PNG" alt="Login Screen" class="inline" height= "100%" width= "100%" /> 
+
+## Proposed solution
+An app can mark that the `performInteraction` is a multiselect type by setting the `InteractionMode` as `MANUAL_MULTISELECT`.  
+
+
+Update the `InteractionMode` enum in the `MOBILE_API`
+```xml
+	<enum name="InteractionMode" since="1.0">
+		.
+		.
+		.
+		<element name="MANUAL_MULTISELECT">
+			<description>This mode causes the interaction to only occur on the display, meaning the choices are provided only via the display. In this mode, the user can select more than 1 item in the perform interaction before pressing a button to confirm all choices. </description>
+	
+	</enum>
+```
+
+Plus the `PerformInteraction` response needs to update from a single item to an array in the `MOBILE_API`
+```xml
+<function name="PerformInteraction" funcitonID="PerformInteractionID" messageType="response" since="1.0">
+	.
+	.
+	.
+	<param name="choiceID" type="Integer" minvalue="0" maxvalue="2000000000" array="true" minsize="1" maxsize="100" mandatory="false" since="3.0">
+		<description>
+			IDs of the choices that were selected i response to PerformInteraction.
+			Only is valid if general result is "success:true".
+		</desciption>
+	</param>
+</function>
+```
+
+and update the `HMI_API`
+```
+<function name="PerformInteraction" messagetype="response">
+	<param name="choiceID" type="Integer" minvalue="0" maxvalue="2000000000" array="true" minsize="1" maxsize="100" mandatory="false" since="3.0">
+		<description>
+			IDs of the choices that were selected i response to PerformInteraction.
+			Only is valid if general result is "success:true".
+		</desciption>
+	</param>
+</function>
+
+```
+
+On older head units, the app might request a multiselect `performInteraction` but the app would get a response of INVALID_DATA. So the app would have to handle business logic while thats going on
```

Comment:
```suggestion
 On older head units, the app might request a multiselect `performInteraction` but the head unit would ignore the parameter of `multiselectConfirmationButtonText` and the response would only end up being a single choiceID.
```
Per item 4 in the [requested revisions](https://github.com/smartdevicelink/sdl_evolution/issues/1052#issuecomment-676497507).

Reply:
I can see that you've updated this line, but it still doesn't match what was requested in the Steering Committee decision. I'll need to verify this change internally and get back to you. If you can share any insight into why your change is different from what was originally requested, that would be very helpful!